widget/app.R

```diff
@@ -0,0 +1,62 @@
+# Copyright(c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+suppressMessages(library(shiny))
+suppressMessages(library(DT))
+
+
+server <- function(input, output, session) {
+  
+  values <- reactiveValues(widget_state = "initializing",
+                           current_run = NULL)
+
+  # retrieve run object once widget_state
+  observeEvent(values$widget_state == "streaming", {
+    ws <- azuremlsdk::get_workspace(ws_name, subscription_id, rg)
+    exp <- azuremlsdk::experiment(ws, exp_name)
+    values$current_run <- azuremlsdk::get_run(exp, run_id)
+  },
+  ignoreInit = TRUE,
+  once = TRUE)
+
+  # stop app if user closes session
+  session$onSessionEnded(function() {
+    shiny::stopApp()
+  })
+  
+  plot <- function() {
+    if (!is.null(values$current_run)) {
+      if (isolate(values$current_run$status) %in% 
+          c("Failed", "Completed", "Canceled")) {
+        print("Your run has reached a terminal state. The widget will close now.")
+        Sys.sleep(3)
+        shiny::stopApp()
```

Comment:
Questions:

- Any reason of sleep 3 secs here?
- What happened to viewer pane after we call stopApp()?

Reply:
1) Because otherwise the app would close down before the user has a chance to read the warning message telling them it's closing. I don't think we want it to be so abrupt that users think the app crashed.
2) No, the viewer pane clears once Shiny stops an app.

Reply:
Can we quietly shutdown the shiny app and switch to the static version widget? User experience should remain same with the current one.

Reply:
As discussed, can we keep the job running but stop keep fetching run details?

Reply:
sure - sorry, i forgot you asked for that change.